packages/math/src/stable/index.ts

```diff
@@ -162,6 +162,45 @@ function _tokenInForExactSpotPriceAfterSwap(
   );
 }
 
+function _tokenOutForExactSpotPriceAfterSwap(
+  spotPrice: OldBigNumber,
+  poolPairData: StablePoolPairData,
+  outGuess = poolPairData.allBalances[poolPairData.tokenIndexOut].times(
+    bnum(0.01)
+  ),
+  spotPricePrecision = bnum(0.0000001),
+  iteration_number = 0
+): OldBigNumber {
+  // Calculate the amount of tokenIn needed to reach the desired spotPrice
+  // The Newton-Raphson method is used to find the SpotPrice of the function
+  // Results could be inaccurate for very small amounts of tokenIn
+  const guessedSpotPrice =
+    StableMaths._spotPriceAfterSwapTokenInForExactTokenOut(outGuess, {
+      ...poolPairData,
+    });
+  const diffFromSpotPrice = spotPrice.minus(guessedSpotPrice);
+  if (diffFromSpotPrice.abs().lte(spotPricePrecision)) {
+    return outGuess;
+  }
+
+  iteration_number += 1;
+  if (iteration_number > 255) {
```

Comment:
how did you define 255?

Reply:
It is the same number to calculate the invariant in the solidity

Reply:
`#should` a comment here it just pointing that out 😬